pkg/operator/apis/monitoring/v1/types.go

```diff
@@ -695,6 +695,21 @@ func endpointScrapeConfig(id, projectID, location, cluster string, ep ScrapeEndp
 		httpCfg.ProxyURL.URL = proxyURL
 	}
 
+	if ep.HTTPClientConfig.TLS != nil {
+		unsupportedFormat := "Setting %q is not supported. See https://github.com/GoogleCloudPlatform/prometheus-engine/issues/450 for more details"
+		if ep.HTTPClientConfig.TLS.CA != nil {
```

Comment:
This is odd a little bitt. We are adding fields we want in future, sure, but we produce documentation and ability to parse this only to tell user that it's not possible to use (since propagation mechanism has to be designed and implemented) - and it's not even documented anywhere, just in error message.

I understand the reusability of it, but wonder if this will not confuse users and upset. Perhaps we could create temporary type with only skip field? (should be easy to replace with TLSConfig later on?)

Reply:
hrm - yea tricky one. My hesitancy with a temporary type is we'll have to handle [CRD versioning](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#overview), which isn't the most fun thing.

You could also create a duplicate type that we slowly add to over time so it's backwards compatible. That seems a little messy, but does avoid dealing with versions/conversions.

At the very least, you could add a comment about lack of support [here](https://github.com/GoogleCloudPlatform/prometheus-engine/pull/454/files#diff-918b2f2eda7a36f6f638775958962c3aeeaa1272fa163f6f4a3633da478f667fR1015), which should make it's way to the API docs with `make presubmit`.

Reply:
Going with the duplicate type idea, since when we generate the CRD metadata, it does not show up as a separate type because it's an inline field. Thanks!